File: web/src/domain/entities/Organization.ts
export interface Organization {
  id: string
  name: string
  visibility: "public" | "private"
  description: string
  logo_url?: string
  owner_id: string
  membership_role?: "owner" | "admin" | "member"
  created_at: string
  updated_at: string
  owner?: {
    id: string
    username?: string
    display_name?: string
    avatar_url?: string
  }
}

export interface CreateOrganizationInput {
  name: string
  description: string
  logo_url?: string
  visibility?: "public" | "private"
}

export interface UpdateOrganizationInput {
  name?: string
  description?: string
  logo_url?: string
  visibility?: "public" | "private"
}

export interface OrganizationInvitation {
  id: string
  organization_id: string
  invited_by: string
  email: string
  role: "admin" | "member"
  status: "pending" | "accepted" | "declined"
  expires_at: string
  created_at: string
  updated_at: string
}

export interface OrganizationMember {
  id: string
  account_id: string
  role: "owner" | "admin" | "member"
  status: "pending" | "accepted" | "declined"
  joined_at: string
  profile?: {
    id: string
    username?: string
    display_name?: string
    avatar_url?: string
  }
}

export interface OrganizationListResponse {
  organizations: Organization[]
  total: number
  limit: number
  offset: number
}

File: api/model/organization.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Organization struct {
	ID             uuid.UUID `gorm:"primaryKey" json:"id"`
	Name           string    `gorm:"uniqueIndex" json:"name"`
	Visibility     string    `gorm:"type:varchar(20);default:public;index" json:"visibility"`
	Description    string    `json:"description"`
	LogoURL        *string   `gorm:"size:2048" json:"logo_url,omitempty"`
	OwnerID        uuid.UUID `gorm:"index" json:"owner_id"`
	MembershipRole *string   `gorm:"->;column:membership_role;-:migration" json:"membership_role,omitempty"`
	Owner          *Account  `json:"owner,omitempty"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}
